core/modules/filters/x-listops.js

```diff
@@ -186,4 +204,33 @@ Extended filter operators to manipulate the current list.
         }, []);
         return set;
     };
+
+	/*
+	Returns the next item from the operand list after any matching item from the current list -- else the first item from the operand list
+	*/
+	exports.cycle = function(source, operator) {
+		var array = $tw.utils.parseStringArray(operator.operand, "true"),
+			results = prepare_results(source),
+			len = array.length,
+			found = 0,
+			p;
+		for(p = 0; p < len; p++) {
```

Comment:
hmm, this code looks complicated. Wouldn't it be easier to read (and imo understand) if we would have `array.getIndex() array.next() array.prev()` functions, that return the value, instead of this complicated 4 level deep ifs?